dimagi/commcare-odk
# PIN auth

cross: https://github.com/dimagi/javarosa/pull/230, https://github.com/dimagi/commcare/pull/243

How to run/test this:
-Install and log into an app, go into Developer Preferences and turn on "Give option to use PIN for login" (this is turned off by default, and individual domains will be able to set a custom field in their app properties on HQ to enable it).
-Log out again. The next time you log in, a dialog will pop up offering you the option to set a PIN now, set one later, or not to ask again. 
-Selecting set a PIN will take you to the pin create screen. On that same screen, the options menu contains a "Remember PW For Next Login" option. If you select this and confirm, you will be logged out immediately, and the next login will be primed.
-At any time (if the PIN setting is turned on), you can also access the PIN create screen via the home screen options menu. If you navigate to the PIN create screen this way, you will first be required to authenticate yourself with either your current PIN if you have one, or your password if not.
